src/main/java/duke/command/AddEventCommand.java

```diff
@@ -0,0 +1,60 @@
+package duke.command;
+
+import duke.task.Event;
+import duke.ui.Ui;
+import duke.exception.DukeException;
+import duke.task.TaskList;
+/**
+ * It is a command object extends from Command for the Duke program.
+ * When the parser calls it, it will receive the requests from the users
+ * during the running of the program and starts add new events to the task list.
+ */
+public class AddEventCommand extends Command {
+
+    /**
+     * Constructor for AddEventCommand object
+     *
+     * @param userMessage The message that the user inputs for further execution.
+     */
+    public AddEventCommand(String userMessage) {
+        super(userMessage);
+    }
+
+    /**
+     * The execution after parsing, it will add a event object into the tasks.
+     * If the input is not correct, it will raise an exception.
+     *
+     * @param taskList The current taskList in the program.
+     * @param ui The current ui in the program.
+     * @throws DukeException if there are some cases such as no event time specified, no event task name, then
+     * it will raise the DukeException.
+     */
+    public void execute(TaskList taskList, Ui ui) throws DukeException {
+        StringBuilder builder = new StringBuilder();
+        builder.append("Got it! I've added this task:\n");
+        int spaceIndex = userMessage.indexOf(" ");
+        int dateIndex = userMessage.indexOf('/');
+        if (dateIndex == -1) {
+            throw new DukeException("OOPS!!! I can't find your event time.");
+        }
+        if (spaceIndex == -1 || dateIndex - spaceIndex == 1) {
+            throw new DukeException("OOPS!!! The description of an event cannot be empty.");
+        }
```

Comment:
I like how you're checking for invalid inputs with very specific exception messages! This way the user would have an easy time telling what's wrong.